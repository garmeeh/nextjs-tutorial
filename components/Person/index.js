import styled from 'styled-components';

const PersonWrapper = styled.div`
  background: #273469;
  color: #fafaff;
  border-radius: 4px;
  overflow: hidden;
  text-align: center;
  box-shadow: -1px 1px 4px 0px rgba(48, 52, 63, 1);
`;

const Header = styled.div`
  padding: 10px 0 0;
`;

const Name = styled.p`
  font-size: 24px;
  font-family: 'Roboto Slab', serif;
`;

const Details = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 10px;
`;

const Detail = styled.p`
  font-size: 13px;
`;

const Films = styled.div`
  padding: 10px;
  text-transform: uppercase;
  font-weight: bold;
  background: #30343f;
`;

const Person = ({ name, birth_year, height, films, mass }) => (
  <PersonWrapper>
    <Header>
      <Name>{name}</Name>
    </Header>
    <Details>
      <Detail>
        <strong>D.O.B:</strong> {birth_year}
      </Detail>
      <Detail>
        <strong>Height:</strong> {height}
      </Detail>
      <Detail>
        <strong>Mass:</strong> {mass}
      </Detail>
    </Details>
    <Films>
      <Detail>
        Appears in {films.length} film{films.length > 1 ? 's' : ''}
      </Detail>
    </Films>
  </PersonWrapper>
);

export default Person;
