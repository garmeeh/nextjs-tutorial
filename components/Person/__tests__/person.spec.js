import React from 'react';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';
import Person from '../';

const mockPerson = {
  name: 'Biggs Darklighter',
  birth_year: '24BBY',
  height: '183',
  films: ['https://swapi.co/api/films/1/']
};

it('matches snapshot', () => {
  const { container } = render(<Person {...mockPerson} />);
  expect(container.firstChild).toMatchSnapshot();
});

it('film is singular', () => {
  const { getByTestId } = render(<Person {...mockPerson} />);
  expect(getByTestId('films').innerHTML).toBe('Appears in 1 film');
});

it('film is pluralised', () => {
  mockPerson.films.push('https://swapi.co/api/films/2/');
  const { getByTestId } = render(<Person {...mockPerson} />);
  expect(getByTestId('films').innerHTML).toBe('Appears in 2 films');
});
