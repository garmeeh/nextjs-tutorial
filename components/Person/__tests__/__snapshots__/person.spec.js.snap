// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="Person__PersonWrapper-s1764vzb-0 cZQQoU"
>
  <div
    class="Person__Header-s1764vzb-1 dIRTXs"
  >
    <p
      class="Person__Name-s1764vzb-2 kquhVD"
    >
      Biggs Darklighter
    </p>
  </div>
  <div
    class="Person__Details-s1764vzb-3 ctxkcF"
  >
    <p
      class="Person__Detail-s1764vzb-4 hgIVdu"
    >
      <strong>
        D.O.B:
      </strong>
       
      24BBY
    </p>
    <p
      class="Person__Detail-s1764vzb-4 hgIVdu"
    >
      <strong>
        Height:
      </strong>
       
      183
    </p>
    <p
      class="Person__Detail-s1764vzb-4 hgIVdu"
    >
      <strong>
        Mass:
      </strong>
       
    </p>
  </div>
  <div
    class="Person__Films-s1764vzb-5 hdmUDC"
  >
    <p
      class="Person__Detail-s1764vzb-4 hgIVdu"
      data-testid="films"
    >
      Appears in 
      1
       film
      
    </p>
  </div>
</div>
`;
